doc/ip-interface/loopback-action/ip-interface-loopback-action-design.md

```diff
@@ -0,0 +1,318 @@
+Intforch performs validity checks, call the relevant SAI API and updates internal cache. There is an existing mechanizm in intforch that postpones handling of changes in INTF_TABLE until interface is ready, we will use it.
+
+## 2.2 DB schema
+### 2.2.1 Config DB
+To support loopback action set on L3 interface, a new field-value pair is defined in CONFIG_DB in the following tables.
+Valid values are "drop" or "forward".
+
+1. VLAN_INTERFACE
+```
+VLAN_INTERFACE|interface-name
+    "loopback_action": action (OPTIONAL) 
+```
+Example:
+```
+"VLAN_INTERFACE": {
+    "Vlan100": {
+        "loopback_action": "drop",
+        "mac_addr": "00:01:02:03:04:10",
+        "mpls": "enable"
+    },
+}
+```
+
+2. INTERFACE
+```
+INTERFACE|interface-name
+    "loopback_action": action 
+```
+Example:
+
+```
+"INTERFACE": {
+    "Ethernet248": {
+        "loopback_action": "forward",
+        "mac_addr": "00:01:02:03:04:11",
+        "mpls": "enable"
+    },
+}
+```
+3. PORTCHANNEL_INTERFACE
+```
+PORTCHANNEL_INTERFACE|interface-name
+    "loopback_action": action 
+```
+Example:
+```
+"PORTCHANNEL_INTERFACE": {
+    "PortChannel1": {
+        "loopback_action": "drop",
+        "mac_addr": "00:01:02:03:04:12",
+        "mpls": "enable"
+    },
+}
+```
+
+### 2.2.2 App DB
+To support loopback action set on L3 interface, a new field-value pair is defined in APP DB in table INTF_TABLE.
+```
+INTF_TABLE|interface-name
+    "loopback_action": action
+```
+Example for interface vlan:
+```
+    "INTF_TABLE": {
+        "Vlan100": {
+            "loopback_action": "drop",
+            "mac_addr": "00:01:02:03:04:10",
+            "mpls": "enable"            
+        }
+    }
+```
+Example for interface Ethernet:
+```
+    "INTF_TABLE": {
+        "Ethernet248": {
+            "loopback_action": "drop",
+            "mac_addr": "00:01:02:03:04:11",
+            "mpls": "enable"
+        }
+    }
+```
+Example for interface port-channel:
+```
+    "INTF_TABLE": {
+        "PortChannel1": {
+            "loopback_action": "drop",
+            "mac_addr": "00:01:02:03:04:12",
+            "mpls": "enable"            
+        }
+    }
+```
+
+### 2.2.3 Configuration migration
+No special handling is required
+
+## 2.3 Flows
+
+### 2.3.1 Loopback action set interface vlan
+#### 2.3.1.1 Handle set in runtime
+![interface vlan](images/interface_vlan.svg "Figure 2: interface vlan")
+###### Figure 2: interface vlan
+
+#### 2.3.1.2 Handle set in init
+In init, loopback action configuration is being handled as part of the IP interface creation.
+During init, all field-value pairs of each key in INTF_TABLE table (e.g. INTF_TABLE|Vlan100) are being received in intfOrch together at once.
+Those field-value pairs are the interface attributes, e.g. mac address, mpls enable, loopback action etc.
```

Comment:
funny things you have used mpls where nvidia does not support it. If you have other field to share it might be better

Reply:
Done.